<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AndroidSettingsWidget</class>
 <widget class="QWidget" name="AndroidSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>843</width>
    <height>625</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Android Configuration</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="OpenJDKLocationLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>JDK location:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="Utils::PathChooser" name="OpenJDKLocationPathChooser" native="true"/>
   </item>
   <item row="1" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="jdkWarningIconLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="jdkWarningLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="2">
    <widget class="QToolButton" name="downloadOpenJDKToolButton">
     <property name="toolTip">
      <string>Download JDK</string>
     </property>
     <property name="icon">
      <iconset resource="android.qrc">
       <normaloff>:/android/images/download.png</normaloff>:/android/images/download.png</iconset>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="SDKLocationLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Android SDK location:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="Utils::PathChooser" name="SDKLocationPathChooser" native="true"/>
   </item>
   <item row="2" column="2">
    <widget class="QToolButton" name="downloadSDKToolButton">
     <property name="toolTip">
      <string>Download Android SDK</string>
     </property>
     <property name="icon">
      <iconset resource="android.qrc">
       <normaloff>:/android/images/download.png</normaloff>:/android/images/download.png</iconset>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="sdkWarningIconLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="sdkWarningLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="NDKLocationLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Android NDK location:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="Utils::PathChooser" name="NDKLocationPathChooser" native="true"/>
   </item>
   <item row="4" column="2">
    <widget class="QToolButton" name="downloadNDKToolButton">
     <property name="toolTip">
      <string>Download Android NDK</string>
     </property>
     <property name="icon">
      <iconset resource="android.qrc">
       <normaloff>:/android/images/download.png</normaloff>:/android/images/download.png</iconset>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <property name="spacing">
      <number>2</number>
     </property>
     <item>
      <widget class="QLabel" name="gdbWarningIconLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="gdbWarningLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>&lt;a href=&quot;xx&quot;&gt;The GDB in the NDK appears to have broken python support.&lt;/a&gt;</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="6" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="spacing">
      <number>2</number>
     </property>
     <item>
      <widget class="QLabel" name="ndkWarningIconLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="toolchainFoundLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="8" column="1">
    <widget class="QCheckBox" name="CreateKitCheckBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Automatically create kits for Android tool chains</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="9" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="spacing">
      <number>2</number>
     </property>
     <item>
      <widget class="QLabel" name="kitWarningIconLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="kitWarningLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="10" column="1">
    <widget class="QCheckBox" name="UseGradleCheckBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Use Gradle instead of Ant</string>
     </property>
    </widget>
   </item>
   <item row="11" column="0">
    <widget class="QLabel" name="AntLocationLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Ant executable:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="11" column="1">
    <widget class="Utils::PathChooser" name="AntLocationPathChooser" native="true"/>
   </item>
   <item row="11" column="2">
    <widget class="QToolButton" name="downloadAntToolButton">
     <property name="toolTip">
      <string>Download Ant</string>
     </property>
     <property name="icon">
      <iconset resource="android.qrc">
       <normaloff>:/android/images/download.png</normaloff>:/android/images/download.png</iconset>
     </property>
    </widget>
   </item>
   <item row="12" column="0" colspan="2">
    <widget class="QFrame" name="AVDManagerFrame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="3" column="1">
       <widget class="QPushButton" name="AVDStartPushButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Start...</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="AVDManagerLabel">
          <property name="text">
           <string>AVD Manager</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="DataPartitionSizeLable">
          <property name="text">
           <string>System/data partition size:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="DataPartitionSizeSpinBox">
          <property name="suffix">
           <string> Mb</string>
          </property>
          <property name="maximum">
           <number>99999</number>
          </property>
          <property name="value">
           <number>1024</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="manageAVDPushButton">
          <property name="text">
           <string>Start AVD Manager...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="AVDRemovePushButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Remove</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="AVDAddPushButton">
        <property name="text">
         <string>Add...</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" rowspan="4">
       <widget class="QTableView" name="AVDTableView">
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="textElideMode">
         <enum>Qt::ElideMiddle</enum>
        </property>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
       </widget>
      </item>
      <item row="4" column="1">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>129</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="android.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>AVDAddPushButton</sender>
   <signal>clicked()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>addAVD()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>793</x>
     <y>394</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AVDRemovePushButton</sender>
   <signal>clicked()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>removeAVD()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>793</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AVDStartPushButton</sender>
   <signal>clicked()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>startAVD()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>793</x>
     <y>460</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AVDTableView</sender>
   <signal>activated(QModelIndex)</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>avdActivated(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>328</x>
     <y>605</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AVDTableView</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>avdActivated(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>328</x>
     <y>605</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DataPartitionSizeSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>dataPartitionSizeEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>614</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>manageAVDPushButton</sender>
   <signal>clicked()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>manageAVD()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>792</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CreateKitCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>createKitToggled()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>543</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>339</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>downloadSDKToolButton</sender>
   <signal>clicked()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>openSDKDownloadUrl()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>downloadNDKToolButton</sender>
   <signal>clicked()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>openNDKDownloadUrl()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>downloadAntToolButton</sender>
   <signal>clicked()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>openAntDownloadUrl()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>313</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>downloadOpenJDKToolButton</sender>
   <signal>clicked()</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>openOpenJDKDownloadUrl()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>29</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>UseGradleCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>AndroidSettingsWidget</receiver>
   <slot>useGradleToggled()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>395</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>sdkLocationEditingFinished()</slot>
  <slot>ndkLocationEditingFinished()</slot>
  <slot>antLocationEditingFinished()</slot>
  <slot>addAVD()</slot>
  <slot>removeAVD()</slot>
  <slot>startAVD()</slot>
  <slot>avdActivated(QModelIndex)</slot>
  <slot>dataPartitionSizeEditingFinished()</slot>
  <slot>openJDKLocationEditingFinished()</slot>
  <slot>manageAVD()</slot>
  <slot>createKitToggled()</slot>
  <slot>openSDKDownloadUrl()</slot>
  <slot>openNDKDownloadUrl()</slot>
  <slot>openAntDownloadUrl()</slot>
  <slot>openOpenJDKDownloadUrl()</slot>
 </slots>
</ui>
